### PR TITLE
Adds i18n for Wizard + add communication between wizard & actions from usecase

### PR DESCRIPTION
**Pull request by ajainarayanan**
### Note:

This PR is branched off from #6759 
DO NOT MERGE THIS PR BEFORE THAT
- Adds i18n for Wizards.
- Adds `AbstractWizard` component to abstract consumers using wizard with just type of wizard and store specified for usage.
- Adds adhoc way of notifying the actions (whether its done or not) and updates UI for actions appropriately.
#### WIP:
- Styling
- Refining how wizards and actions interact with each other
- Upload data to streams in stream wizard
- Deploy a pipeline with just JSON and navigate to respective detailed view of pipeline in hydrator.

### CODE REVIEW

**Comment by ajainarayanan** [comment]
Closing the PR and opening the same against `feature/cdap-4-sneak-peek`

File: cdap-ui/app/cdap/components/CaskWizards/StreamCreate/UploadStep/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <Provider store={CreateStreamStore}>
+      <div className="wizard-upload-step">
+        <div className="well well-lg">
+          <Dropzone
+            onDrop={onFileDrop}
+            className="drop-container"
+            activeClassName="on-drag-over-container"
+          >
+            <span className="file-drop-placeholder">
+              Select file to be uploaded to Stream
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CaskWizards/StreamCreate/UploadStep/index.js:37]: I18n?

File: cdap-ui/app/cdap/components/CaskWizards/StreamCreate/index.js [hunk 2]
```diff
@@ -51,14 +65,15 @@ export default class StreamCreateWizard extends Component {
         {
           this.state.showWizard ?
             <WizardModal
-              title="Create Stream"
+              title={this.props.context ? this.props.context + " | Create Stream" : "Create Stream"}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CaskWizards/StreamCreate/index.js]: i18n

File: cdap-ui/app/cdap/components/Wizard/index.js [hunk 3]
```diff
@@ -157,6 +149,39 @@ export default class Wizard extends Component{
...
+                  return {
+                    index,
+                    label: step.shorttitle,
+                    complete: this.props.store.getState()[step.id] && this.props.store.getState()[step.id].__complete
+                  };
+                })
+                .map((value) => {
+                  return (
+                    <li key={shortid.generate()}>
+                      Step - {value.index} : {value.label} - {value.complete ? 'Done' : 'Skipped'}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Wizard/index.js]: I18n

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 4]
```diff
@@ -68,7 +68,14 @@ features:
...
       useCases: Use Cases
-
+    action-types:
+      create_stream:
+        name: Create Stream
+        icon: icon-streams
+      create_app:
+        name: Create App
+        icon: icon-app
+      create_artifact: Create Artifact
```
Comment (elfenheart) [inline cdap-ui/app/cdap/text/text-en.yaml]: We should order these alphabetically so it's easy to navigate

**Review by elfenheart — CHANGES_REQUESTED** [review]

File: cdap-ui/app/cdap/components/Wizard/index.js [hunk 5]
```diff
@@ -157,6 +149,39 @@ export default class Wizard extends Component{
       let state = this.props.store.getState()[stepId];
       return state && state.__complete;
     };
+    const onSubmitSuccessMessage = () => {
+      return (
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/Wizard/index.js:156]: we should avoid assigning function to const 
ref: https://github.com/airbnb/javascript#functions

File: cdap-ui/app/cdap/components/CaskWizards/StreamCreate/UploadStep/index.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/CaskWizards/StreamCreate/UploadStep/index.js:37]: Ignore this. This is getting removed in the next PR.

File: cdap-ui/app/cdap/components/Wizard/index.js (hunk 5 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/Wizard/index.js:156]: Hhmm it doesn't say avoid using const. It says use named functions. Ok will do.

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
Will create a separate PR for update based on namespace changes. Merging this for now

**Pull request merged**